packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -106,151 +283,140 @@ function WorklistDataGrid(props) {
       return;
     }
-
-    // Remove the ability to receive visible focus as only one cell should be able to receive tab focus.
-    // This also ensures that tabbing back into the grid focuses on the cell that last had focus.
-    grid.current.rows[currentFocusedRow].cells[currentFocusedCol].tabIndex = -1;
-
-    focusedRow.current = nextRow;
-    focusedCol.current = nextCol;
-
-    const elementWithTabStop = grid.current.rows[nextRow].cells[nextCol];
-    elementWithTabStop.tabIndex = 0;
-    elementWithTabStop.focus();
-    event.preventDefault();
+    handleMoveCellFocus(cellCoordinates, { row: nextRow, col: nextCol });
+    event.preventDefault(); // prevent the page from moving with the arrow keys.
   };
 
-  const handleClick = (event) => {
-    // Determine which cell was clicked. In the event that the user holds the mouse across multiple cells,
-    // the originating cell is the clicked cell/active element.
-    const clickedCell = event.target.closest('td,th') || document.activeElement.closest('td,th');
-
-    // Ensure that the cell is focusable and is a valid cell
-    if (!clickedCell) {
-      return;
-    }
-    // Remove Tab stop from previous cell in table that has focus and set it to the cell that was clicked.
-    grid.current.rows[focusedRow.current].cells[focusedCol.current].tabIndex = -1;
-    focusedRow.current = clickedCell.parentElement.rowIndex;
-    focusedCol.current = clickedCell.cellIndex;
-
-    clickedCell.tabIndex = 0;
-    clickedCell.focus();
-    event.preventDefault();
-  };
-
-  const getCellData = (cell, cellColumnIndex) => {
-    const tabIndex = { tabIndex: '-1' };
-
-    // Determine whether cell is a header or grid cell
-    const WorklistCellTag = props.rowHeaderIndex === cellColumnIndex ? 'th' : 'td';
-    let cellAriaLabel;
-
-    if (cell.isMasked) {
-      cellAriaLabel = intl.formatMessage({ id: 'Terra.worklistDataGrid.maskedCell' });
-    } else if (!cell.content) {
-      cellAriaLabel = intl.formatMessage({ id: 'Terra.worklistDataGrid.blank' });
-    }
-
-    return (
-      // Return worklist data grid cell component
-      <WorklistCellTag
-        key={cellColumnIndex}
-        {...tabIndex}
-        className={cx('worklist-data-grid-cell', { masked: cell.isMasked, selectable: !(cell.isMasked || cell.isSelectable === false), blank: !cell.content })}
-        aria-label={cellAriaLabel}
-      >
-        {!cell.isMasked && cell.content
-          && <div className={cx('cell-content')}>{cell.content}</div>}
-      </WorklistCellTag>
-    );
-  };
-
-  const buildColumn = (column) => (
+  const buildColumn = (column, columnIndex) => (
     <ColumnHeaderCell
+      key={column.id}
       column={column}
-      width={column.width || props.columnWidth}
+      width={props.columnWidth}
       headerHeight={columnHeaderHeight}
-      onColumnSelect={onColumnSelect}
+      onColumnSelect={handleColumnSelect}
+      coordinates={{ row: 0, col: columnIndex }}
```

Comment:
This logic should cause each column header cell to render every time because a new object is being created each time for the coordinates.  You will most likely have to separate this object into individual properties so that the primitive values can be compared.

Reply:
Rather than use one object for the prop, I have two props with the individual parts.